### PR TITLE
Pass delegatecall flag into precompile RunAndCalculateGas

### PR DESCRIPTION
**Pull request by jewei1997**

### CODE REVIEW

File: core/vm/contracts.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ type PrecompiledContract interface {
 }
```
Comment (codchen) [inline core/vm/contracts.go:44]: let's also add this flag here^

File: core/vm/evm.go [hunk 2]
```diff
@@ -310,6 +311,7 @@ func (evm *EVM) CallCode(caller ContractRef, addr common.Address, input []byte,
 // DelegateCall differs from CallCode in the sense that it executes the given address'
 // code with the caller as context and the caller is set to the caller of the caller.
 func (evm *EVM) DelegateCall(caller ContractRef, addr common.Address, input []byte, gas uint64) (ret []byte, leftOverGas uint64, err error) {
+	fmt.Println("JEREMYDEBUG: In DelegateCall, caller = ", caller.Address(), " addr = ", addr)
```
Comment (codchen) [inline core/vm/evm.go:314]: nit remove

File: core/vm/evm.go [hunk 3]
```diff
@@ -332,7 +334,8 @@ func (evm *EVM) DelegateCall(caller ContractRef, addr common.Address, input []by
 		// NOTE: caller must, at all times be a contract. It should never happen
 		// that caller is something other than a Contract.
 		parent := caller.(*Contract)
-		ret, gas, err = RunPrecompiledContract(p, evm, parent.CallerAddress, caller.Address(), input, gas, nil, evm.Config.Tracer, evm.interpreter.readOnly)
+		fmt.Println("JEREMYDEBUG: calling precompile contract")
```
Comment (codchen) [inline core/vm/evm.go:337]: ditto

File: core/vm/contracts.go (hunk 1 above)
Comment (jewei1997) [reply core/vm/contracts.go:44]: fixed

**Review by codchen — APPROVED** [review]

**Review by codchen — APPROVED** [review]

**Pull request merged**